components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -3512,6 +3517,111 @@ private void deleteRequestPathAuthenticators(int applicationID, Connection conne
+            throw new IdentityApplicationManagementException("Error while retrieving trusted app configurations.", e);
+        }
+        return spTrustedAppMetadata;
+    }
+
+    /**
+     * Update trusted app metadata of the application.
+     *
+     * @param applicationId  Application ID.
+     * @param trustedAppMetadata Trusted app configurations.
+     * @param connection     Database connection.
+     * @throws IdentityApplicationManagementException If an error occurs while updating trusted app configurations.
+     */
+    private void updateSpTrustedAppMetadata(int applicationId, SpTrustedAppMetadata trustedAppMetadata,
+                                            Connection connection) throws IdentityApplicationManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Adding trusted app configurations for application: " + applicationId);
+        }
+        int tenantID = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+
+        try (PreparedStatement storeAppConfigs = connection
+                    .prepareStatement(ApplicationMgtDBQueries.STORE_TRUSTED_APPS)) {
+            if (trustedAppMetadata != null) {
+                storeAppConfigs.setInt(1, applicationId);
+                storeAppConfigs.setBoolean(5, trustedAppMetadata.getIsFidoTrusted());
+                storeAppConfigs.setInt(6, tenantID);
+
+                if (StringUtils.isNotBlank(trustedAppMetadata.getAndroidPackageName())) {
+                    storeAppConfigs.setString(2, ANDROID);
+                    storeAppConfigs.setString(3, trustedAppMetadata.getAndroidPackageName());
+                    storeAppConfigs.setString(4, trustedAppMetadata.getAndroidThumbprints());
+                    storeAppConfigs.addBatch();
+                }
+                if (StringUtils.isNotBlank(trustedAppMetadata.getAppleAppId())) {
+                    storeAppConfigs.setString(2, IOS);
+                    storeAppConfigs.setString(3, trustedAppMetadata.getAppleAppId());
+                    storeAppConfigs.setString(4, null);
+                    storeAppConfigs.addBatch();
+                }
+                storeAppConfigs.executeBatch();
+            }
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementException("Error while storing trusted app configurations.", e);
+        }
+    }
+
+    /**
+     * Delete trusted app metadata of the application.
+     *
+     * @param applicationID Application ID.
+     * @param connection   Database connection.
+     * @throws SQLException If an error occurs while deleting trusted app configurations.
+     */
+    private void deleteSpTrustedAppMetadata(int applicationID, Connection connection) throws SQLException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Deleting trusted app configurations for application: " + applicationID);
+        }
+        int tenantID = CarbonContext.getThreadLocalCarbonContext().getTenantId();
```

Comment:
Instead of resolving the tenantID here lets pass it in as a param. The place where initial calling method is used already has the tenantDomain.